packages/devtools_shared/lib/src/server/server_api.dart

```diff
@@ -207,23 +208,62 @@ class ServerApi {
 
       // ----- Extensions api. -----
 
-      case apiServeAvailableExtensions:
+      case ExtensionsApi.apiServeAvailableExtensions:
```

Comment:
This `switch` statement is ~220 lines now and should probably be refactored (this doesn't have to block this PR, but consider applying these changes to the new APIs for now). A few things that could be changed:

- Almost every request creates a variable for `queryParams`, so it's probably just worth declaring before the switch statement
- Each API endpoint should have its own method. Ideally, the switch statement will only contain a single function call for each case
- I'd create a helper function for creating bad request responses for missing query parameters since they're all basically the same format
- Going even further, if there's a check we perform multiple times, we could also use a helper to perform the check (e.g., `bool checkRootPath(Map<String, Object?> queryParams)` or `bool checkRootPath(String? rootPath)`)
- It might also be worth creating a `FutureOr<shelf.Response> encodeResponse(Object? response) => api.getCompleted(jsonEncode(response))`

Reply:
good ideas. Implemented for the two extension related apis and added a TODO for the rest.